scripts/elasticsearch_nodes.py

```diff
@@ -0,0 +1,107 @@
+"""
+This script is referenced by TerraForm as an external data source. It prints the
+current deployment's Elasticsearch node IDs as JSON to STDOUT. If the list of
+node IDs is less than the number of placeholders in the Cloudwatch dashboard
+template, the output is padded with 'blank' entries.
+Example output:
+{"nodes": "tVsGH2BSXXYe9sMLEYRcIv,ydG9meZMAGkTnAlPgLljz8,blank,blank"}
+"""
+from itertools import (
+    repeat,
+)
+import json
+import logging
+import re
+from time import (
+    sleep,
+)
+
+from botocore.exceptions import (
+    ClientError,
+)
+import jq
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+# How many times to retry the request to AWS if an error is encountered.
+max_attempts = 10
+
+
+def main() -> None:
+    for _ in range(max_attempts):
+        try:
+            response = aws.cloudwatch.list_metrics(Namespace='AWS/ES',
+                                                   MetricName='CPUUtilization',
+                                                   RecentlyActive='PT3H',
+                                                   Dimensions=[
+                                                       {
+                                                           'Name': 'DomainName',
+                                                           'Value': config.es_domain
+                                                       }
+                                                   ])
+        except ClientError as e:
+            log.debug('Error: ' + e.response['Error']['Code'] + '. Retrying...')
+        else:
+            nodes = [
+                d['Value']
+                for m in response['Metrics']
+                for d in m['Dimensions']
+                if d['Name'] == 'NodeId'
+            ]
+            if len(nodes) > 0:
+                break
+            else:
+                log.debug('Error: No node IDs found in response.')
+        sleep(5)
+    else:
+        raise RuntimeError('Failed to obtain node IDs. Max attempts exhausted.')
+
+    # The list of nodes IDs is appended with dummy entries to ensure the length
+    # matches the number of placeholders in the template. If our list had less
+    # than the template, Terraform would throw an "Invalid index" error. Also,
+    # the dummy entries cannot be empty strings.
+    placeholder_count = node_placeholder_count()
+    require(placeholder_count > 0,
+            'Error: No node ID placeholders found in template file.')
+    nodes = sorted(nodes)
+    nodes += list(repeat('blank', placeholder_count - len(nodes)))
+
+    # As required by Terraform for an external data source, we output the data
+    # as a map of string values.
+    result = {
+        "nodes": ','.join(nodes)
+    }
+    print(json.dumps(result))
```

Comment:
```suggestion
    return {
        "nodes": ','.join(nodes)
    }
```

and move `print(json.dumps(result))` to the top level.

Reply:
Might also want to rename `main` to a more descriptive name.